File: frontend/src/types/merchantPgMapping.ts
export enum MerchantPgMappingStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE'
}

export const MERCHANT_PG_MAPPING_STATUS_LABELS: Record<MerchantPgMappingStatus, string> = {
  [MerchantPgMappingStatus.ACTIVE]: '활성',
  [MerchantPgMappingStatus.INACTIVE]: '비활성'
};

export interface MerchantPgMappingDto {
  id: string;
  merchantId: string;
  merchantName?: string;
  merchantCode?: string;
  pgConnectionId: string;
  pgCode?: string;
  pgName?: string;
  mid: string;
  terminalId?: string;
  catId?: string;
  config?: Record<string, unknown>;
  status: MerchantPgMappingStatus;
  createdAt: string;
  updatedAt: string;
}

export interface MerchantPgMappingCreateRequest {
  merchantId: string;
  pgConnectionId: string;
  mid: string;
  terminalId?: string;
  catId?: string;
  config?: Record<string, unknown>;
  status?: MerchantPgMappingStatus;
}

export interface MerchantPgMappingUpdateRequest {
  pgConnectionId?: string;
  mid?: string;
  terminalId?: string;
  catId?: string;
  config?: Record<string, unknown>;
  status?: MerchantPgMappingStatus;
}

File: src/main/java/com/korpay/billpay/domain/entity/MerchantPgMapping.java
package com.korpay.billpay.domain.entity;

import com.korpay.billpay.domain.enums.MerchantPgMappingStatus;
import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Type;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Table(name = "merchant_pg_mappings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class MerchantPgMapping {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "merchant_id", nullable = false)
    private Merchant merchant;

    @Column(name = "pg_connection_id", nullable = false)
    private Long pgConnectionId;

    @Column(name = "mid", nullable = false, length = 50)
    private String mid;

    @Column(name = "terminal_id", length = 50)
    private String terminalId;

    @Column(name = "cat_id", length = 50)
    private String catId;

    @Type(JsonBinaryType.class)
    @Column(name = "config", columnDefinition = "jsonb")
    private Map<String, Object> config;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    private MerchantPgMappingStatus status;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;
}
